Use Lucenes expectThrows() when testing exceptions

cbuescher opened the pull request:
Cleaning up a few remaining occurences of using junits ExpectedException rule in favor of using LuceneTestCase#expectThrows() which is more concise and versatile.

nik9000 reviewed (approved):
Do you want to add ExpectedException to the forbidden-apis?

cbuescher commented:
I think it was rarely used and think its not necessary to stricly enforce this, I just think expectThrows()  is much nicer. So no, I wasn't thinking of forbidden-apis, unless you think its important to do so.

nik9000 commented:
> So no, I wasn't thinking of forbidden-apis, unless you think its important to do so.

I don't think it is super important but it might be a nice reminder for someone who isn't familiar with the code base.

cbuescher commented:
@nik9000 I was hesitating to add this to forbidden-apis because its not a real error and I thought there's no good way of communicating the "other way" as best practice in a good way, but now I see forbidden-apis provides some way of letting the user know what to do instead, so I added ExpectedException it with a pointer LuceneTestCase#expectThrows(). Anything else to change from your side?

cbuescher committed:
Use Lucenes expectThrows() when testing exceptions

Cleaning up a few remaining occurences of using junits ExpectedException rule in
favor of using LuceneTestCase#expectThrows() which is more concise and versatile.

cbuescher commented:
Also merged to 5.x with 257f4fd7
